src/crimson/osd/osd.cc

```diff
@@ -800,11 +756,10 @@ OSD::ShardDispatcher::ms_dispatch(
       // FIXME: order is not guaranteed in this path
       return conn.get_foreign(
       ).then([this, m=std::move(m)](auto f_conn) {
-        return container().invoke_on(PRIMARY_CORE,
-            [f_conn=std::move(f_conn), m=std::move(m)]
-            (auto& local_dispatcher) mutable {
+        return seastar::smp::submit_to(PRIMARY_CORE,
+            [f_conn=std::move(f_conn), m=std::move(m), this]() mutable {
           auto conn = make_local_shared_foreign(std::move(f_conn));
-          return local_dispatcher.ms_dispatch(conn, std::move(m));
+          return *do_ms_dispatch(conn, std::move(m));
```

Comment:
```
auto ret = do_ms_dispatch(conn, std::move(m));
ceph_assert(ret.has_value());
return do_ms_dispatch(conn, std::move(m)).value();
```

Reply:
Will use assert(ret.has_value()) to document that it must be handled in this method.